ui/components/src/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { IconButton, Dialog as MuiDialog, DialogTitle, DialogContent, DialogActions, Button } from '@mui/material';
+import CloseIcon from 'mdi-material-ui/Close';
+
+export interface DialogProps {
+  isOpen: boolean;
+  title: string;
+  onClose: () => void;
+  primaryButton: DialogButtonProps;
+  secondaryButton: DialogButtonProps;
+  children: React.ReactNode;
+}
+
+export interface DialogButtonProps {
+  name: string;
+  onClick: () => void;
+}
+
+const DIALOG_DEFAULT_WIDTH = 500;
+
+export const Dialog = ({ isOpen, title, children, onClose, primaryButton, secondaryButton }: DialogProps) => {
```

Comment:
Just a general thought, but I know we're debating a bit whether we want to really be building a "component library" for general UI components and this kinda feels like that. Should we hold off on making this a component for now until we think about that some more?

For this component in particular, every time we've tried to create a single generic "Dialog" component in our internal codebase, we've basically ended up getting rid of it because it's really hard to build one that covers all use cases. We've typically ended up going back to something more composable like Dialog, DialogTitle, DialogActions, etc. (i.e. what's in MUI or some slightly custom version of it). I think that might be the other reason to hold off on doing this now.

Anyway, just a couple general things I was thinking about when I skimmed this real quick. Happy to chat more if you want to talk it through!

Reply:
makes sense, I removed the generic dialog component